primitives/src/merkle_tree/universal_merkle_tree.rs

```diff
@@ -285,6 +299,30 @@ mod mt_tests {
             .unwrap()
             .is_ok());
         }
+        for i in 0..10 {
+            mt.update_with(F::from(i as u64), |elem| match elem {
+                Some(elem) => Some(*elem),
+                None => Some(F::from(i as u64)),
+            })
+            .unwrap();
+        }
+        assert_eq!(mt.num_leaves(), 10);
+        // test lookup at index 7
+        let (val, proof) = mt.universal_lookup(F::from(7u64)).expect_ok().unwrap();
+        assert_eq!(val, &F::from(7u64));
+        assert_eq!(proof.elem().unwrap(), val);
+        assert!(
+            RescueSparseMerkleTree::<F, F>::verify(&mt.root.value(), F::from(7u64), &proof)
+                .unwrap()
+                .is_ok()
+        );
+
+        // Remove index 8
+        mt.update_with(F::from(8u64), |_| None).unwrap();
+        assert!(mt
+            .universal_lookup(F::from(8u64))
+            .expect_not_found()
+            .is_ok());
```

Comment:
should we also check other `mt.num_leaves()` is also correctly updated?

Reply:
added [bd2bc19](https://github.com/EspressoSystems/jellyfish/pull/475/commits/bd2bc19dcc34a7b0647794c15cc820ff0fa27c79)